packages/flutter/test/widgets/transform_test.dart

```diff
@@ -388,6 +388,119 @@ void main() {
     },
     skip: isBrowser, // due to https://github.com/flutter/flutter/issues/42767
   );
+
+  testWidgets('Transform.translate with FilterQuality produces filter layer', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Transform.translate(
+        offset: const Offset(25.0, 25.0),
+        child: const SizedBox(width: 100, height: 100),
+        filterQuality: FilterQuality.low,
+      ),
+    );
+    expect(tester.layers.whereType<ImageFilterLayer>().length, 1);
+  });
+
+  testWidgets('Transform.scale with FilterQuality produces filter layer', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Transform.scale(
+        scale: 3.14159,
+        child: const SizedBox(width: 100, height: 100),
+        filterQuality: FilterQuality.low,
+      ),
+    );
+    expect(tester.layers.whereType<ImageFilterLayer>().length, 1);
+  });
+
+  testWidgets('Transform.rotate with FilterQuality produces filter layer', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Transform.rotate(
+        angle: math.pi / 4,
+        child: const SizedBox(width: 100, height: 100),
+        filterQuality: FilterQuality.low,
+      ),
+    );
+    expect(tester.layers.whereType<ImageFilterLayer>().length, 1);
+  });
+
+  testWidgets('Transform layers update to match child and filterQuality', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Transform.rotate(
+        angle: math.pi / 4,
+        child: const SizedBox(width: 100, height: 100),
+        filterQuality: FilterQuality.low,
+      ),
+    );
+    expect(tester.layers.whereType<ImageFilterLayer>().length, 1);
+
+    await tester.pumpWidget(
+      Transform.rotate(
+        angle: math.pi / 4,
+        child: const SizedBox(width: 100, height: 100),
+      ),
+    );
+    expect(tester.layers.whereType<ImageFilterLayer>().length, 0);
```

Comment:
nit: for testing zero length consider using the `isEmpty` matcher, also for better error messages (e.g. "expected empty, but found 5 elements")